pkg/cri/server/helpers.go

```diff
@@ -242,6 +242,29 @@ func (c *criService) ensureImageExists(ctx context.Context, ref string, config *
 	return &newImage, nil
 }
 
+// validateTargetContainer checks that a container is a valid
+// target for a container using PID NamespaceMode_TARGET.
+// The target container must be in the same sandbox and must be running.
+// Returns the target container for convenience.
+func (c *criService) validateTargetContainer(sandboxID string, nsOpts *runtime.NamespaceOption) (containerstore.Container, error) {
+	targetContainerID := nsOpts.TargetId
+	targetContainer, err := c.containerStore.Get(targetContainerID)
+	if err != nil {
+		return containerstore.Container{}, errors.Wrapf(err, "container %q does not exist", targetContainerID)
+	}
+
+	if targetContainer.Metadata.SandboxID != sandboxID {
+		return containerstore.Container{}, errors.Errorf("container %q belongs to a different sandbox", targetContainerID)
```

Comment:
Do we want to add the sandbox id too?

Reply:
For this one maybe, maybe not for the one where the target container doesn't exist at all.

```
container %q (sandbox %s) does not belong to sandbox %s
```
Something like that? It feels a little redundant but at least it has all the information in.

Reply:
more is definitely better in this case, esp. because it's already an odd add on debug container use case :-)

Reply:
Alright, I've changed it to what I suggested.